### DISCUSSION TITLE
Audiokit Delay works in one example but not another

### DISCUSSION THREAD

**Discussion by coatsiecoates** [pschatzmann/arduino-audio-tools#1478] [Q&A] [answered]
Using an A1S 8388 audiokit and ArduinoIDE 2.3.2, in the the example "streams-synthbasic1-audiokit.ino", I added a delay and it worked perfectly.

when I added the very same four lines in the very same places in the example "streams-stk-audiokit", and rerouted the final StreamCopy to 'effects' rather than 'in', I get:

_" 'dely' was not declared in this scope"_

the #includes are the same, except the AudioSTK was not in "streams-synthbasic1-audiokit.ino", which worked.

Do I need to call effects differently if using the STK?

And how do I access the STK effects like STKFreeverb? I could only get the Delay to work in "streams-synthbasic1-audiokit.ino", but not the STKEcho.

Sorry if this is obvious, but it seems there's a lot of reorganizing on the site. I scoured for examples over two days but got a lot of 404's during this update.

in the following example, the placement of all the commented lines are in the same places as in the working example but I get errors now. If I comment out the added lines and return StreamCopy copier(out, effects) to StreamCopy copier(out, in), I get an operational sketch. but of course, with no effect:

```
#include "AudioTools.h"
#include "AudioLibs/AudioSTK.h"
#include "AudioLibs/AudioBoardStream.h"

Sitar instrument(440);
STKStream<Instrmnt> in(instrument);
AudioBoardStream out(AudioKitEs8388V1);
AudioEffectStream effects(in);              //////////////////////////////       ADDED    ///////////////////////////// ///////////////////////////// 
Delay dely(1000, 0.5, 0.33, 44100, false);  //////////////////////////////       ADDED     ///////////////////////////// ///////////////////////////// 
StreamCopy copier(out, effects);  //////////////////////////////////     MODIFIED TO "EFFECTS"  RATHER THAN "IN" ///////////////////////////// 
MusicalNotes notes;

float note_amplitude = 0.5;
static float notes_array[] = {
  // frequencies aleatoric C-scale
  N_C3, N_D3, N_E3, N_F3, N_G3, N_A3, N_B3,
  N_C4, N_D4, N_E4, N_F4, N_G4, N_A4, N_B4,
  N_C4, N_D4, N_E5, N_F4, N_G4, N_A4, N_B5
};

void play() {
  static bool active = true;
  static float freq;
  static uint64_t timeout;

  if (millis() > timeout) {
    if (active) {
      // play note for 800 ms
      int count = sizeof(notes_array) / sizeof(float);

      freq = notes_array[random(count)];  // generate random number from 0 to count-1

      instrument.noteOn(freq * 0.25, note_amplitude);

      timeout = millis() + 150;
      active = false;
    } else {
      // silence for 100 ms
      instrument.noteOff(note_amplitude);
      timeout = millis() + 15;
      active = true;
    }
  }
}

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);

  effects.addEffect(dely);                   //////////////////////////////    ADDED    ///////////////////////////// ///////////////////////////// 
  // setup input
  auto icfg = in.defaultConfig();
  in.begin(icfg);
  effects.begin(icfg); //////////////////  ADDED USING INPUT CONFIG. "CFG" WAS USED IN THE WORKING EXAMPLE  ////////////////////////// 
  // setup output
  auto ocfg = out.defaultConfig(TX_MODE);
  ocfg.copyFrom(icfg);
  ocfg.sd_active = false;
  out.begin(ocfg);
}

void loop() {
  play();
  copier.copy();
}
```

**Comment by pschatzmann** [comment, marked as answer]
Just look at the error message: with your include of STK there are 2 Delay classes now and you need to tell the system which one to take by using the namespace! audio_tools::Delay or stk::Delay ?

audio_tools::Delay dely(1000, 0.5, 0.33, 44100, false);

**Reply by coatsiecoates** [reply]
I did try that before asking, and it compiled, but there was no audio.

**Reply by coatsiecoates** [reply]
my sample rate was incorrect in the delay setup. Thanks for swatting me back into the namespace, I had already given up on it.
